File: medplum-java/medplum-model/src/main/java/com/fhir/MeasurereportType.java
package com.fhir;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Locale;
import javax.annotation.Nonnull;

public final class MeasurereportType {
  public static final MeasurereportType SUMMARY = new MeasurereportType(Value.SUMMARY, "SUMMARY");

  public static final MeasurereportType INDIVIDUAL = new MeasurereportType(Value.INDIVIDUAL, "INDIVIDUAL");

  private final Value value;

  private final String string;

  MeasurereportType(Value value, String string) {
    this.value = value;
    this.string = string;
  }

  public Value getEnumValue() {
    return value;
  }

  @Override
  @JsonValue
  public String toString() {
    return this.string;
  }

  @Override
  public boolean equals(Object other) {
    return (this == other) 
      || (other instanceof MeasurereportType && this.string.equals(((MeasurereportType) other).string));
  }

  @Override
  public int hashCode() {
    return this.string.hashCode();
  }

  public <T> T accept(Visitor<T> visitor) {
    switch (value) {
      case SUMMARY:
        return visitor.visitSUMMARY();
      case INDIVIDUAL:
        return visitor.visitINDIVIDUAL();
      case UNKNOWN:
      default:
        return visitor.visitUnknown(string);
    }
  }

  @JsonCreator(
      mode = JsonCreator.Mode.DELEGATING
  )
  public static MeasurereportType valueOf(@Nonnull String value) {
    String upperCasedValue = value.toUpperCase(Locale.ROOT);
    switch (upperCasedValue) {
      case "SUMMARY":
        return SUMMARY;
      case "INDIVIDUAL":
        return INDIVIDUAL;
      default:
        return new MeasurereportType(Value.UNKNOWN, upperCasedValue);
    }
  }

  public enum Value {
    INDIVIDUAL,

    SUMMARY,

    UNKNOWN
  }

  public interface Visitor<T> {
    T visitINDIVIDUAL();

    T visitSUMMARY();

    T visitUnknown(String unknownType);
  }
}

File: medplum-typescript/src/model/fhir/MeasurereportType.ts
export type MeasurereportType =
    | MeasurereportType.Individual
    | MeasurereportType.Summary;

export const MeasurereportType = {
    Individual: "individual" as MeasurereportType.Individual,
    Summary: "summary" as MeasurereportType.Summary,

    _visit: <Result>(value: MeasurereportType, visitor: MeasurereportType._Visitor<Result>): Result => {
        switch (value) {
            case MeasurereportType.Individual: return visitor.individual();
            case MeasurereportType.Summary: return visitor.summary();
            default: return visitor._unknown();
        }
    },
};

export declare namespace MeasurereportType {
    export type Individual = "individual" & {
        __Individual: void,
    };
    export type Summary = "summary" & {
        __Summary: void,
    };

    export interface _Visitor<Result> {
        individual: () => Result;
        summary: () => Result;
        _unknown: () => Result;
    }
}
